framework/tst/dslabs/framework/testing/newviz/SingleNodePanel.java

```diff
@@ -147,14 +160,32 @@ void updateState(EventTreeState s, boolean viewDeliveredMessages) {
             if (messages.containsKey(message)) {
                 continue;
             }
-            addMessage(message);
+            addMessage(message,
+                       !searchDone && (settings == null || settings.shouldDeliver(message)));
             repaintMessageBox = true;
         }
         for (MessageEnvelope message : new HashSet<>(messages.keySet())) {
             if (!ms.contains(message)) {
                 messageBox.remove(messages.get(message).getLeft());
                 repaintMessageBox = true;
                 messages.remove(message);
+            } else {
+                // Update messages if they are not deliverable. Slightly gross because we have to
```

Comment:
No need to do that. I think we can change `messages` to `Map<MessageEnvelope, Pair<StateTree, JButton>>` or something. The `JPanel` is always their direct parent.

Reply:
Fixed in https://github.com/emichael/dslabs/pull/29/commits/472418acfffa20d1d968e3e5a293784bd37e2bb0.